src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -234,9 +234,23 @@ public Task CallFunctionAsync(string functionName, params object[] parameters)
 
             // TODO: Support for versioning
             string version = DefaultVersion;
-            this.config.AssertActivityExists(functionName, version);
+            FunctionType functionType = this.config.GetFunctionType(functionName, version);
 
-            Task<TResult> callTask = this.innerContext.ScheduleTask<TResult>(functionName, version, parameters);
+            Task<TResult> callTask;
+
+            switch (functionType)
+            {
+                case FunctionType.Activity:
+                    callTask = this.innerContext.ScheduleTask<TResult>(functionName, version, parameters);
+                    break;
+                case FunctionType.Orchestrator:
+                    callTask = this.innerContext.CreateSubOrchestrationInstance<TResult>(functionName, version, JsonConvert.SerializeObject(parameters));
+                    break;
+                default:
+                    throw new ArgumentException(
+                        string.Format("The function '{0}' doesn't exist, is disabled, or is not an activity or orchestrator function",
+                            functionName));
```

Comment:
The point of this default clause is to flag the case where we see a function type enum value that we don't expect. Therefore, the correct exception is going to be something like throw new InvalidOperationException($"Unexpected function type {functionType}.");